indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
 
-    def _processContribution(self,contrib,l):
-        if not contrib.canAccess(self._aw):
+    def _processContribution(self, contrib, l):
+        if not contrib.can_access(self._aw):
             return
-        date = "%s"%escape(contrib.getAdjustedStartDate(self._tz).strftime("%H:%M"))
-        #date="<font size=\"" + str(modifiedFontSize(10, self._fontsize)) + "\">%s</font>"%date
-        #date=self._fontify(date,10)
-        #date.style=self._styles["table_body"]
-        date=Paragraph(date,self._styles["table_body"])
-        lt=[]
-        captionText="[%s] %s"%(escape(contrib.getId()),escape(contrib.getTitle()))
+
+        lt = []
+        date = Paragraph(format_time(contrib.start_dt, timezone=self._tz), self._styles["table_body"])
+        caption = '[{}] {}'.format(contrib.id, escape(contrib.title))
+
         if not self._ttPDFFormat.showContribId():
-            captionText="%s"%(escape(contrib.getTitle()))
-        if self._ttPDFFormat.showLengthContribs():
-            captionText="%s (%s)"%(captionText, escape((datetime(1900,1,1)+contrib.getDuration()).strftime("%Hh%M'")))
-        if self._ttPDFFormat.showContribAbstract():
-            captionText="<font name=\"Times-Bold\"><b>%s</b></font>"%captionText
-        captionText="<font size=\"" + str(modifiedFontSize(10, self._fontsize)) + "\">%s</font>"%captionText
-        lt.append([self._fontify(captionText,10)])
-        colorCell = ""
+            caption = escape(contrib.title)
+        elif self._ttPDFFormat.showLengthContribs():
+            caption = "{} ({})".format(caption, format_human_timedelta(contrib.timetable_entry.duration))
+        elif self._ttPDFFormat.showContribAbstract():
+            caption = '<font name="Times-Bold"><b>{}</b></font>'.format(caption)
+
+        color_cell = ""
+        caption = '<font size="{}">{}</font>'.format(str(modifiedFontSize(10, self._fontsize)), caption)
+        lt.append([self._fontify(caption, 10)])
+
         if self._useColors():
-            colorCell = " "
+            color_cell = " "
         if self._ttPDFFormat.showContribAbstract():
-            spkList=[]
-            for spk in contrib.getSpeakerList():
-                spkName = spk.getFullName()
+            speaker_list = []
+            for spk in contrib.speakers:
+                person = spk.person
+                speaker_name = person.get_full_name(last_name_first=False, show_title=True)
                 if not self._ttPDFFormat.showSpeakerTitle():
-                    spkName = self._getNameWithoutTitle(spk)
-                if self._showSpeakerAffiliation and spk.getAffiliation().strip() != "":
-                    spkName += " (" + spk.getAffiliation() + ")"
-                spkList.append(spkName)
-            if len(spkList) > 0:
-                if len(spkList) == 1:
-                    speakerWord = i18nformat(""" _("Presenter"): """)
+                    speaker_name = self._getNameWithoutTitle(person)
```

Comment:
Wouldn't this be the same as `spkList = [spk.get_full_name(last_name_first=False, show_title= self._ttPDFFormat.showSpeakerTitle()) for spk in contrib.speakers]` (not tested)?
Note that we're calling `get_full_name(...)` directly on the link, which is the correct way to do it.